stl/inc/vector

```diff
@@ -227,7 +227,7 @@ template <class _Myvec>
 struct pointer_traits<_Vector_const_iterator<_Myvec>> {
     using pointer         = _Vector_const_iterator<_Myvec>;
     using element_type    = const typename pointer::value_type;

```

Comment:
This `typename` still seems to be very "up".

Reply:
This is by design - the presence of the `const` means that it's embedded within a type, not appearing at the top level. (At least, that's my understanding of the rule, and MSVC certainly rejects it.)

Reply:
I believe `value_type` is "the terminal name of a _type-specifier_ of a _defining-type-id_", in which case it is in a type-only context per [temp.res]/4.2.2. That said, removing `typename` is such a low-value change that I'm not sure it's worth our time to file a compiler bug.

Reply:
GCC and Clang seem to agree: https://godbolt.org/z/YzYvYvq3P.

Reply:
Fine, I'll file a bug. Guilt me into it, why don't you.

Reply:
This is VSO-1854115 "C++20 Down With Typename is confused by cv-qualifiers".